src/contract/assignments.rs

```diff
@@ -394,8 +395,43 @@ impl AutoConceal for Assignments {
     }
 }
 
-impl CommitEncodeWithStrategy for Assignments {
-    type Strategy = commit_strategy::UsingStrict;
+impl CommitEncode for Assignments {
+    fn commit_encode<E: io::Write>(self, e: E) -> usize {
+        fn vector_commit_encode<T, E>(v: Vec<T>, e: E) -> usize
+        where
+            T: CommitEncode,
+            E: io::Write,
+        {
+            use lnpbp::client_side_validation::{merklize, MerkleNode};
+            use crate::bitcoin_hashes::Hash;
+
+            merklize(
```

Comment:
Yes, I am aware of that. However, for us to be able to use it we would need a change to be made to the `client_side_validation` crate. As a matter of fact, I have already proposed that change, via PR [#178](https://github.com/LNP-BP/rust-lnpbp/pull/178) of the `LNP-BP/rust-lnpbp` repository.

So, instead of waiting for the upstream upgrade, I have decided to "replicate" the implementation of the CommitEncode trait for vectors of a generic type, with the intention to replace it once the proposed `client_side_validation` crate change is accepted.